api/src/env.ts

```diff
@@ -362,7 +365,7 @@ const typeMap: Record<string, string> = {
 let env: Record<string, any> = {
 	...defaults,
 	...process.env,
-	...processConfiguration(),
+	...(await processConfiguration()),
```

Comment:
This one await will impact all other files that import this to also become asynchronous.
This already hinders us from bundling Directus into a single executable. Recommend to encase this in an exported function instead.